zerver/tests/test_email_change.py

```diff
@@ -1,22 +1,31 @@
 import datetime
 
 from django.core import mail
+from django.urls import reverse
 from django.utils.timezone import now
 
-from confirmation.models import Confirmation, confirmation_url, generate_key
+from confirmation.models import Confirmation, confirmation_url, generate_key, get_object_from_key
 from zerver.lib.actions import do_set_realm_property, do_start_email_change_process
+from zerver.lib.initial_password import initial_password
 from zerver.lib.test_classes import ZulipTestCase
-from zerver.models import (
-    EmailChangeStatus,
-    Realm,
-    UserProfile,
-    get_realm,
-    get_user,
-    get_user_by_delivery_email,
-    get_user_profile_by_id,
-)
+from zerver.models import EmailChangeStatus, Realm, UserProfile, get_user_by_delivery_email
 
 
+def get_email_change_confirmation_url(user_profile: UserProfile, new_email: str, key: str,
+                                      email_address_visibility: int=Realm.EMAIL_ADDRESS_VISIBILITY_EVERYONE,
+                                      ) -> str:
```

Comment:
With the previous test helper, this is now adding the 3rd place in the code with the same aim of creating email change confirmation url, we don't want that. `email_address_visibility` seems like a redundant arg here?

Reply:
Agreed. And all the checks which were that the login page was loaded were also redundant as they are already tested in `EmailChangeTestCase`. So, I've imported `create_email_change_confirmation_key` and just used the key to send POST requests.

Yes, `email_address_visibility ` is redundant. Seems like I forgot to add code specific to that. I'm surprised lint didn't complain. (removed that function now.)